content/sips/sip-366.md

```diff
@@ -16,36 +16,277 @@ proposal: >-
+ * @param collateralType The address of the collateral type that the
+ * account intends to delegate
+ * @param newCollateralAmountD18 The new amount of collateral that
+ * the account intends to delegate, denominated with 18 decimals of
+ * precision
+ * @param leverage The new intended leverage used in the position
+ */
+function declareIntentToDelegateCollateral(
+    uint128 accountId,
+    uint128 poolId,
+    address collateralType,
+    uint256 newCollateralAmountD18,
+    uint256 leverage
+) public;
+```
+
+The `declareIntentToDelegateCollateral` function should, at minimum, satisfy the following requirements:
+
+1. Ensure the caller is authorized to represent the account.
+2. Verify the account holds enough collateral to execute the intent.
+3. Check the validity of the collateral amount to be delegated, considering the `escrowedCollateralByAccount`.
+4. Update the `escrowedCollateralByAccount` to reflect the collateral amount declared in the intent for the identified account and pool.
+5. Update the `intentNoncesByAccount` to include the new intent nonce associated with the account.
+6. Update the `intentByNonce` to represent the newly declared intent using the specific nonce.
+7. Emit a `DelegateCollateralIntentDeclared` event.
+
+#### Processing Intents
+
+Add the following function to **process** an outstanding intent to delegate a new amount of collateral (refer to the NatSpec annotations for further implementation guidance):
+
+```solidity
+/**
+ * @notice Attempt to process an outstanding intent to delegate a
+ * new amount of collateral
+ * @dev The {msg.sender} can be any address
+ * @param accountId The ID of the account for which the outstanding
+ * intent to delegate a new amount of collateral is being processed
+ * @param poolId The ID of the pool for which the intent of the account
+ * to delegate a new amount of collateral is being processed
+ * @param collateralType The address of the collateral type that the
+ * account has an outstanding intent to delegate a new amount of
+ */
+function processIntentToDelegateCollateral(
+    uint128 accountId,
+    uint128 poolId,
+    address collateralType
+) public;
+```
+
+The `processIntentToDelegateCollateral` function should, at minimum, satisfy the following requirements:
+
+1. Allow anyone to call it without requiring permissions, enabling the processing of any open intent.
+2. Verify the existence of the specified intent for the given account, pool, and type of collateral.
+3. Ensure the intent's submission timestamp is greater than the current timestamp plus the applicable delay.
+4. Ensure the intent's submission timestamp does not surpass the current timestamp plus both the delay and window period.
+5. Check that processing the intent won't cause the collateralization ratio to fall below the required issuance ratio.
+6. Update the `escrowedCollateralByAccount` to mirror the amount of collateral in process for the designated account and pool.
```

Comment:
Goes without saying, that if a user pushed an intent to delegate, then he should be blocked from withdrawing that delegated collateral. As if the collateral had been delegated. Does this make sense? And should we add this to the workflow, as it might need to alter core?

Reply:
added section